run.ps1

```diff
@@ -0,0 +1,34 @@
+# set the base tools directory
+$toolsLocalPath = Join-Path (Get-Location) "Tools"
+$runExe = Join-Path $toolsLocalPath "Microsoft.DotNet.BuildTools.Run\run.exe"
+
+# if the tool doesn't exist then download and run the boot-strapper script
+if ((Test-Path $runExe) -eq 0)
+{
+    if ((Test-Path $toolsLocalPath) -eq 0)
+    {
+        mkdir $toolsLocalPath | Out-Null
+    }
+
+    # download boot-strapper script
+    $bootStrapperPath = Join-Path $toolsLocalPath "bootstrap.ps1"
+    Invoke-WebRequest "https://raw.githubusercontent.com/dotnet/buildtools/master/bootstrap/bootstrap.ps1" -OutFile $bootStrapperPath
+
+    # now execute it
+    Invoke-Expression "$bootStrapperPath (Get-Location) $toolsLocalPath" | Out-File (Join-Path (Get-Location) "bootstrap.log")
+    if ($LastExitCode -ne 0)
+    {
+        Write-Output "Boot-strapping failed with exit code $LastExitCode"
+        exit $LastExitCode
+    }
+
+    # copy nuget.exe into the packages directory (this is unique to the build tools repo)
+    $packagesPath = Join-Path (Get-Location) "packages"
+    Get-ChildItem (Join-Path $packagesPath "NuGet.CommandLine") -filter "nuget.exe" -recurse | foreach { copy $_.FullName $packagesPath }
```

Comment:
Agreed that avoiding the need for custom code is best.  If we're ok with adding complexity to .toolversions we could include source and destination directories.  Something like this.

```
Microsoft.DotNet.BuildTools,1.0.26-prerelease-00802-01,lib,.tools
```

So this would translate to "copy the contents in lib to the .tools directory".  It alleviates the need for extra scripts but does add complexity/duplication and internal knowledge of packages to this file.